src/linux/ebpf.cpp

```diff
@@ -99,6 +99,48 @@ Try<int> load(const Program& program)
   return *fd;
 }
 
+
+// Detach an eBPF program from a target (AKA path) by its attachment type
+// and program id. Returns Nothing() on success or if no program is found.
+Try<Nothing> detach(uint32_t attach_type, int program_id, const string& target)
+{
+  Try<int> target_fd = os::open(target, O_DIRECTORY | O_RDONLY | O_CLOEXEC);
+  if (target_fd.isError()) {
+    return Error("Failed to open '" + target + "': " + target_fd.error());
+  }
+
+  bpf_attr attr;
+  memset(&attr, 0, sizeof(attr));
+  attr.prog_id = program_id;
+
+  Try<int, ErrnoError> program_fd =
+    bpf(BPF_PROG_GET_FD_BY_ID, &attr, sizeof(attr));
```

Comment:
from the docs, it looks like loading a program produces an fd, and unloading a program is done by closing the fd, so you probably want to keep that in mind here and figure out the implications of that (do we need an unload function that will detach and close fds..?, replacement will need to close the previous fd..? etc)

Reply:
Noted. Updated to close this `program_fd` file descriptor to have `detach()` effectively detach and unload.